proxmox/nodes/vms/vms_types.go

```diff
@@ -1233,6 +1241,21 @@ func (r CustomStorageDevices) EncodeValues(_ string, v *url.Values) error {
 	return nil
 }
 
+// EncodeValues converts a CustomTPMState struct to a URL vlaue.
+func (r CustomTPMState) EncodeValues(key string, v *url.Values) error {
+	values := []string{
+		fmt.Sprintf("file=%s", r.FileVolume),
+	}
+
+	if r.Version != nil {
```

Comment:
I'm a bit puzzled about this. When, in my terraform program, I just use `tpm_state {}`, it seems this is being `nil` because what end's up being created in proxmox is a `v1.2` TPM device, but shouldn't this be `v2.0` instead (because that's the default value set in the terraform schema definition)?

Reply:
When in doubt, I always destroy everything and delete the tfstate, but it still does not work as I expected.

I do see the same diff as you, with `v2.0`, but all the requests/responses to/from proxmox do not mention `version=v2.0` at all, in fact, there is no `version` argument floating around at all, so the default proxmox tpm version of `v1.2` is used. And I still don't known why this is happening.

Reply:
Hm... I'll try to debug it today.

Reply:
I forgot to mention:

* I'm using terraform 1.6.4
* I'm using proxmox 8.1.3
* I'm using https://github.com/rgl/terraform-proxmox-debian-example to clone an UEFI debian base image that was previously created by https://github.com/rgl/debian-vagrant#proxmox-usage

Reply:
Ah, it's about cloning then! :) Fixed a small bug in `vmGetTPMState`, should be working now.

BTW, this is a bare minimum template I used for testing TPM, takes a second to apply:
```terrafom
resource "proxmox_virtual_environment_vm" "ubuntu_vm_template" {
  name      = "453-test-template"
  node_name = "pve"
  vm_id     = 453

  started  = false
  template = true
}


resource "proxmox_virtual_environment_vm" "ubuntu_vm_cloned" {
  depends_on = [proxmox_virtual_environment_vm.ubuntu_vm_template]
  name       = "453-test-tpm"
  node_name  = "pve"
  vm_id      = 454

  clone {
    vm_id = 453
  }

  tpm_state {
    datastore_id = "local-lvm"
    version      = "v2.0"
  }

  started = false
}
```

Reply:
thank you! its now working nicely!

I've rebased the branch over v0.39.0. the final PR should be squashed if you agree.

I will do the remaining tasks that you've asked initially.